deepmd/pt/model/task/fitting.py

```
@@ -520,6 +548,13 @@
     ) -> dict[str, torch.Tensor]:
         # cast the input to internal precsion
         xx = descriptor.to(self.prec)
+        nf, nloc, nd = xx.shape
+
+        if self.numb_fparam > 0 and fparam is None:
+            # use default fparam
+            assert self.default_fparam_tensor is not None
```

Ruff (0.12.2)

Use of `assert` detected

(S101)